src/sentry/migrations/0886_split_discover_dataset_saved_queries.py

```diff
@@ -0,0 +1,109 @@
+    """
+    This targets transaction-like data from the split from discover.
+    """
+
+    TYPES = [
+        (DISCOVER, "discover"),
+        (ERROR_EVENTS, "error-events"),
+        (TRANSACTION_LIKE, "transaction-like"),
+    ]
+    TYPE_NAMES = [t[1] for t in TYPES]
+
+
+class DatasetSourcesTypes(Enum):
+    """
+    Ambiguous queries that haven't been or couldn't be categorized into a
+    specific dataset.
+    """
+
+    UNKNOWN = 0
+    """
+     Dataset inferred by either running the query or using heuristics.
+    """
+    INFERRED = 1
+    """
+     Canonical dataset, user explicitly selected it.
+    """
+    USER = 2
+    """
+     Was an ambiguous dataset forced to split (i.e. we picked a default).
+    """
+    FORCED = 3
+
+    @classmethod
+    def as_choices(cls) -> tuple[tuple[int, str], ...]:
+        return tuple((source.value, source.name.lower()) for source in cls)
+
+
+def split_discover_dataset_saved_queries(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    DiscoverSavedQuery = apps.get_model("sentry", "DiscoverSavedQuery")
+    queryset = DiscoverSavedQuery.objects.filter(
+        dataset=DiscoverSavedQueryTypes.DISCOVER,
+    ).select_related("organization")
+
+    for saved_query in RangeQuerySetWrapperWithProgressBar(queryset):
+        try:
+            _get_and_save_split_decision_for_query(saved_query, dry_run=False)
+        except Exception:
+            saved_query.dataset = DiscoverSavedQueryTypes.ERROR_EVENTS
+            saved_query.dataset_source = DatasetSourcesTypes.FORCED.value
+
+
+def reverse_split_discover_dataset_saved_queries(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    DiscoverSavedQuery = apps.get_model("sentry", "DiscoverSavedQuery")
+    DiscoverSavedQuery.objects.all().update(
+        dataset=DiscoverSavedQueryTypes.DISCOVER, dataset_source=DatasetSourcesTypes.UNKNOWN.value
+    )
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = True
```

Comment:
We've already done this for SaaS and ST through
the custom job runner, so this migration does not/should not run on sentry SaaS.
This is purely to get the migration to run for self-hosted